### PR TITLE
Extract a RemoteClusters component from LogicalReplicationService

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

To encapsulate some of the connection management.
It should make the integration of the PG tunneling a bit easier.

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java [hunk 1]
```diff
@@ -178,17 +178,6 @@ static RemoteConnectionStrategy buildStrategy(String clusterAlias,
         }
     }
 
-    public static boolean isConnectionEnabled(Settings connectionSettings) {
```
Comment (mfussenegger) [inline server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java:181]: Didn't like the implicit removal of connections if the settings change to have an empty hosts list. I'd rather we only remove connections if subscriptions are dropped, and enforce that active subscriptions do need non-empty hosts.

**Review by mkleen — APPROVED** [review]
Nice, back to the (Completable)Future! 👍

**Comment by mfussenegger** [comment]
retest this please

Jenkins connectivity errors

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java [hunk 2]
```diff
@@ -286,30 +288,30 @@ private void updateBatchFetched(boolean success,
...
-                        }
-                    )
-                ),
-                e -> {
-                }
-            )
+            () -> {
+                shardReplicationService.getRemoteClusterClient(shardId.getIndex())
+                    .whenComplete((client, err) -> {
+                        RetentionLeaseHelper.renewRetentionLease(
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:294]: shouldn't this only be done if `err == null`?

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java [hunk 3]
```diff
@@ -363,15 +365,14 @@ public void close() throws IOException {
...
-                shardId,
-                clusterName,
-                client,
-                ActionListener.wrap(() -> {})
-            ),
-            e -> {}
-        );
+        shardReplicationService.getRemoteClusterClient(shardId.getIndex())
+            .whenComplete((client, err) -> {
+                RetentionLeaseHelper.attemptRetentionLeaseRemoval(
```
Comment (seut) [inline server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:370]: same here, shouldn't we only renew if no error happened?

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java (hunk 2 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:294]: Right, thanks for catching.

File: server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java (hunk 3 above)
Comment (mfussenegger) [reply server/src/main/java/io/crate/replication/logical/ShardReplicationChangesTracker.java:370]: Changed to use `thenAccept`

**Pull request merged**